app/services/the_tvdb_service.rb

```diff
@@ -0,0 +1,124 @@
+class TheTvdbService
+  BASE_URL = 'https://api.thetvdb.com'.freeze
+  API_KEY = ENV['THE_TVDB_API_KEY'].freeze
+
+  attr_accessor :media_id
+
+  def initialize
+    @token = api_token
+  end
+
+  # This will update all mappings to have a season number
+  # This should be run ONLY ONCE, unless we import data from somewhere else.
+  def update_mapping_seasons!
+    return if mapping_data.blank?
+
+    mapping_data['Anime'].each do |media_ids|
+      # set the media_id (this is for our database)
+      self.media_id = media_ids[0]
+      series_id = media_ids[1]['thetvdb/series'][1]
+      season_id = media_ids[1]['thetvdb/season'][1] if media_ids[1].try(:[], 'thetvdb/season')
+
+      response = get(build_episode_path(series_id, season_id))
+
+      unless response.code == 200
+        # will be deleted the mappings if their is a 404 response code
```

Comment:
Wait, what?

Reply:
How about something like this?

```ruby
next if response.code == 404
raise 'TVDB Error' unless response.success?

# Code to create new mapping
```

Reply:
truthfully, now that we changed it to merge into 1, I can do that because if they don't exist it won't create the new mapping. At the end of the day you are deleting the season and series mapping.